lib/dslash_wilson.cu

```diff
@@ -35,13 +40,25 @@ namespace quda
     inline WilsonApply(ColorSpinorField &out, const ColorSpinorField &in, const GaugeField &U, double a,
                        const ColorSpinorField &x, int parity, bool dagger, const int *comm_override, TimeProfile &profile)
     {
-      constexpr int nDim = 4;
-      WilsonArg<Float, nColor, nDim, recon> arg(out, in, U, a, x, parity, dagger, comm_override);
-      Wilson<decltype(arg)> wilson(arg, out, in);
+      if( in.Ndim() == 4 ){
+        constexpr int nDim = 4;
+        WilsonArg<Float, nColor, nDim, recon> arg(out, in, U, a, x, parity, dagger, comm_override);
+        Wilson<decltype(arg)> wilson(arg, out, in);
+
+        dslash::DslashPolicyTune<decltype(wilson)> policy(
+          wilson, const_cast<cudaColorSpinorField *>(static_cast<const cudaColorSpinorField *>(&in)), in.VolumeCB(),
+          in.GhostFaceCB(), profile);
+      } else {
+        // support for two-flavour fields (or fields with general fifth dim, where a Wilson Dslash
+        // is to be applied to each 4D subspace)
+        constexpr int nDim = 5;
+        WilsonArg<Float, nColor, nDim, recon> arg(out, in, U, a, x, parity, dagger, comm_override);
+        Wilson<decltype(arg)> wilson(arg, out, in);
 
-      dslash::DslashPolicyTune<decltype(wilson)> policy(
-        wilson, const_cast<cudaColorSpinorField *>(static_cast<const cudaColorSpinorField *>(&in)), in.VolumeCB(),
-        in.GhostFaceCB(), profile);
+        dslash::DslashPolicyTune<decltype(wilson)> policy(
+          wilson, const_cast<cudaColorSpinorField *>(static_cast<const cudaColorSpinorField *>(&in)),
+          in.getDslashConstant().volume_4d_cb, in.getDslashConstant().ghostFaceCB, profile);
+      }
```

Comment:
IIRC, the reason is to be found in `include/kernels/dslash_ndeg_twisted_clover_preconditioned.cuh`, where I call `applyWilson` on a 5D spinor. This is different than in the `dslash_ndeg_twisted_mass_preconditioned.cuh` case, where the two *input* flavours are taken care of by different threads and the output is kept in shared memory.

I didn't know how to achieve this call of `WilsonApply` onto a 5D spinor inside of `nDegTwistedCloverPreconditioned` otherwise but I presume that it would be possible to modify this to work in the same way as `nDegTwistedMassPreconditioned` with, I think, the drawback that the clover field would have to be loaded twice (once for each flavour).